File: JSBASIC/171_freq_count.js
// 171	171_freq_count.js	Count the frequency of each element in an array using an object.

let arr = [1, 2, 3, 2, 1, 3, 3, 4];
let freq = {};
for (let i = 0; i < arr.length; i++) {
    freq[arr[i]] = (freq[arr[i]] || 0) + 1;
}
console.log("Array:", arr);
console.log("Frequency:");
for (let key in freq) {
    console.log(key + " => " + freq[key] + " times");
}

File: JavaBasic/171_freq_count.java
// 171	171_freq_count.java	Count the frequency of each element in an array using an object.
import java.util.*;
public class _171_freq_count {
    public static void main(String[] args) {
        int[] arr = {1,2,3,2,1,3,3,4};
        Map<Integer,Integer> freq = new LinkedHashMap<>();
        for (int x:arr) freq.put(x, freq.getOrDefault(x,0)+1);
        for (Map.Entry<Integer,Integer> e : freq.entrySet())
            System.out.println(e.getKey() + " => " + e.getValue() + " times");
    }
}
